### PR TITLE
Adjust testing of the max amount of threads for the snapshot thread pool

### LINKED ISSUE #90336: [CI] ScalingThreadPoolTests testScalingThreadPoolConfiguration failing
This seems related to PR https://github.com/elastic/elasticsearch/pull/90214 so marking it for same labels & team.

**Build scan:**
https://gradle-enterprise.elastic.co/s/5ugkfx2zhq7ss/tests/:server:test/org.elasticsearch.threadpool.ScalingThreadPoolTests/testScalingThreadPoolConfiguration

**Reproduction line:**
`./gradlew ':server:test' --tests "org.elasticsearch.threadpool.ScalingThreadPoolTests.testScalingThreadPoolConfiguration" -Dtests.seed=40C96C7CDBEE82D3 -Dtests.locale=et-EE -Dtests.timezone=America/Argentina/Catamarca -Druntime.java=17`

**Applicable branches:**
main

**Reproduces locally?:**
Didn't try

**Failure history:**
https://gradle-enterprise.elastic.co/scans/tests?tests.container=org.elasticsearch.threadpool.ScalingThreadPoolTests&tests.test=testScalingThreadPoolConfiguration

**Failure excerpt:**
```
java.lang.AssertionError: 
Expected: <5>
     but: was <10>

  at __randomizedtesting.SeedInfo.seed([40C96C7CDBEE82D3:6D4F65F3EDFB6C80]:0)
  at org.hamcrest.MatcherAssert.assertThat(MatcherAssert.java:18)
  at org.junit.Assert.assertThat(Assert.java:956)
  at org.junit.Assert.assertThat(Assert.java:923)
  at org.elasticsearch.threadpool.ScalingThreadPoolTests.lambda$testScalingThreadPoolConfiguration$0(ScalingThreadPoolTests.java:105)
  at org.elasticsearch.threadpool.ScalingThreadPoolTests.runScalingThreadPoolTest(ScalingThreadPoolTests.java:433)
  at org.elasticsearch.threadpool.ScalingThreadPoolTests.testScalingThreadPoolConfiguration(ScalingThreadPoolTests.java:88)
  at jdk.internal.reflect.NativeMethodAccessorImpl.invoke0(NativeMethodAccessorImpl.java:-2)
  at jdk.internal.reflect.NativeMethodAccessorImpl.invoke(NativeMethodAccessorImpl.java:77)
  at jdk.internal.reflect.DelegatingMethodAccessorImpl.invoke(DelegatingMethodAccessorImpl.java:43)
  at java.lang.reflect.Method.invoke(Method.java:568)
  at com.carrotsearch.randomizedtesting.RandomizedRunner.invoke(RandomizedRunner.java:1758)
  at com.carrotsearch.randomizedtesting.RandomizedRunner$8.evaluate(RandomizedRunner.java:946)
  at com.carrotsearch.randomizedtesting.RandomizedRunner$9.evaluate(RandomizedRunner.java:982)
  at com.carrotsearch.randomizedtesting.RandomizedRunner$10.evaluate(RandomizedRunner.java:996)
  at com.carrotsearch.randomizedtesting.rules.StatementAdapter.evaluate(StatementAdapter.java:36)
  at org.apache.lucene.tests.util.TestRuleSetupTeardownChained$1.evaluate(TestRuleSetupTeardownChained.java:44)
  at org.apache.lucene.tests.util.AbstractBeforeAfterRule$1.evaluate(AbstractBeforeAfterRule.java:43)
  at org.apache.lucene.tests.util.TestRuleThreadAndTestName$1.evaluate(TestRuleThreadAndTestName.java:45)
  at org.apache.lucene.tests.util.TestRuleIgnoreAfterMaxFailures$1.evaluate(TestRuleIgnoreAfterMaxFailures.java:60)
  at org.apache.lucene.tests.util.TestRuleMarkFailure$1.evaluate(TestRuleMarkFailure.java:44)
  at com.carrotsearch.randomizedtesting.rules.StatementAdapter.evaluate(StatementAdapter.java:36)
  at com.carrotsearch.randomizedtesting.ThreadLeakControl$StatementRunner.run(ThreadLeakControl.java:390)
  at com.carrotsearch.randomizedtesting.ThreadLeakControl.forkTimeoutingTask(ThreadLeakControl.java:843)
  at com.carrotsearch.randomizedtesting.ThreadLeakControl$3.evaluate(ThreadLeakControl.java:490)
  at com.carrotsearch.randomizedtesting.RandomizedRunner.runSingleTest(RandomizedRunner.java:955)
  at com.carrotsearch.randomizedtesting.RandomizedRunner$5.evaluate(RandomizedRunner.java:840)
  at com.carrotsearch.randomizedtesting.RandomizedRunner$6.evaluate(RandomizedRunner.java:891)
  at com.carrotsearch.randomizedtesting.RandomizedRunner$7.evaluate(RandomizedRunner.java:902)
  at org.apache.lucene.tests.util.AbstractBeforeAfterRule$1.evaluate(AbstractBeforeAfterRule.java:43)
  at com.carrotsearch.randomizedtesting.rules.StatementAdapter.evaluate(StatementAdapter.java:36)
  at org.apache.lucene.tests.util.TestRuleStoreClassName$1.evaluate(TestRuleStoreClassName.java:38)
  at com.carrotsearch.randomizedtesting.rules.NoShadowingOrOverridesOnMethodsRule$1.evaluate(NoShadowingOrOverridesOnMethodsRule.java:40)
  at com.carrotsearch.randomizedtesting.rules.NoShadowingOrOverridesOnMethodsRule$1.evaluate(NoShadowingOrOverridesOnMethodsRule.java:40)
  at com.carrotsearch.randomizedtesting.rules.StatementAdapter.evaluate(StatementAdapter.java:36)
  at com.carrotsearch.randomizedtesting.rules.StatementAdapter.evaluate(StatementAdapter.java:36)
  at org.apache.lucene.tests.util.TestRuleAssertionsRequired$1.evaluate(TestRuleAssertionsRequired.java:53)
  at org.apache.lucene.tests.util.AbstractBeforeAfterRule$1.evaluate(AbstractBeforeAfterRule.java:43)
  at org.apache.lucene.tests.util.TestRuleMarkFailure$1.evaluate(TestRuleMarkFailure.java:44)
  at org.apache.lucene.tests.util.TestRuleIgnoreAfterMaxFailures$1.evaluate(TestRuleIgnoreAfterMaxFailures.java:60)
  at org.apache.lucene.tests.util.TestRuleIgnoreTestSuites$1.evaluate(TestRuleIgnoreTestSuites.java:47)
  at com.carrotsearch.randomizedtesting.rules.StatementAdapter.evaluate(StatementAdapter.java:36)
  at com.carrotsearch.randomizedtesting.ThreadLeakControl$StatementRunner.run(ThreadLeakControl.java:390)
  at com.carrotsearch.randomizedtesting.ThreadLeakControl.lambda$forkTimeoutingTask$0(ThreadLeakControl.java:850)
  at java.lang.Thread.run(Thread.java:833)

```

### PR DESCRIPTION
**Pull request by arteam**
#90282 set the max amount of threads for the snapshot thread pool to be 10 regardless of the amount of processors.

Fixes #90336

### CODE REVIEW

**Review by idegtiarenko — APPROVED** [review]

**Comment by arteam** [comment]
@elasticmachine update branch

File: server/src/test/java/org/elasticsearch/threadpool/ScalingThreadPoolTests.java [hunk 1]
```diff
@@ -73,7 +73,7 @@ public void testScalingThreadPoolConfiguration() throws InterruptedException {
             expectedMax = randomIntBetween(Math.max(1, core), 16);
             builder.put("thread_pool." + threadPoolName + ".max", expectedMax);
         } else {
-            expectedMax = maxBasedOnNumberOfProcessors;
+            expectedMax = threadPoolName.equals("snapshot") ? 10 : maxBasedOnNumberOfProcessors;
```
Comment (original-brownbear) [inline server/src/test/java/org/elasticsearch/threadpool/ScalingThreadPoolTests.java:76]: Maybe use the thread pool name constant here? `ThreadPool.Names.SNAPSHOT`?

**Review by original-brownbear — APPROVED** [review]

**Comment by DaveCTurner** [comment]
I muted the failing test in 3df396e4b717afb430dd2b012404f25d43fbb034 (it's reliably failing and blocking other testing). +1 to Armin's suggestion to use the constant rather than a literal here, and also please unmute the test in this PR.

**Comment by arteam** [comment]
Thanks Ievgen, Armin, and David!

**Pull request merged**